python/python/ert/geo/geo_region.py

```diff
@@ -41,10 +45,68 @@ def __init__(self, pointset, preselect=False):
             super(GeoRegion, self).__init__(c_ptr)
         else:
             raise ValueError('Could not construct GeoRegion from pointset %s.' % pointset)
+        self.select_polygon_fn = {
+            (True, True): self._select_inside_polygon,
+            (False, True): self._select_outside_polygon,
+            (True, False): self._deselect_inside_polygon,
+            (False, False): self._deselect_outside_polygon
+        }
+        self.select_halfspace_fn = {
+            (True, True): self._select_above_line,
+            (False, True): self._select_below_line,
+            (True, False): self._deselect_above_line,
+            (False, False): self._deselect_below_line
+        }
+
 
     def getActiveList(self):
         return self._get_index_list()
 
+    def select_polygon(self, polygon, inside=True, select=True):
+        if not isinstance(polygon, CPolyline):
+            raise ValueError('Need to select with a CPolyline, not %s.'
+                             % type(polygon))
+        selector_fn = self.select_polygon_fn[(inside, select)]
+        selector_fn(polygon)
+
+    def select_halfspace(self, line, above=True, select=True):
+        try:
+            p1, p2 = line
+            x1, y1 = map(float, p1)
+            x2, y2 = map(float, p2)
+        except Exception as err:
+            err_msg = 'Select with pair ((x1,y1), (x2,y2)), not %s (%s).'
+            raise ValueError(err_msg % (line, err))
+        x1x2_ptr = cpair(x1, x2)
+        y1y2_ptr = cpair(y1, y2)
+        selector_fn = self.select_halfspace_fn[(above, select)]
+        selector_fn(x1x2_ptr, y1y2_ptr)  # note that it takes xx, yy
+
+    def select_inside(self, polygon):
```

Comment:
At least for the polygon the  gain from having the `(bool, bool)` tuple based function table is quite limited; these functions could just call the select function directly:

```
def select_inside(self , polygon):
       self._select_inside_polygon( polygon )
```
possibly with a @assert_polygon decorator[1]? Or  just leave it as it is if you think that is best.

[1]: Yes - I think decorators is a quite cool concept which we do not use very much.